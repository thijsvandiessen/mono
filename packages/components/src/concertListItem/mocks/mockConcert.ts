import type { Concert } from '@mono/data'

export const mockConcert: Concert = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [
    {
      startTime: '2023-09-01T13:32:36+01:00',
      id: 'test id',
      title: 'test',
      address: 'test',
      ticketLink: 'https://example.com',
      lat: 0,
      lng: 0,
    },
  ],
  url: 'some-slug',
  content: [],
}
