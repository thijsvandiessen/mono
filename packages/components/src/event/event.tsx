import React from 'react'
import { EventListItem } from '../eventListItem'
import { getEvent } from '@mono/data'

export interface Props {
  className?: string
  id: string
  size?: 'small' | 'large'
  isLast?: boolean
}

export const Event = async ({
  className,
  id,
  size = 'small',
  isLast,
}: Props) => {
  const { data } = await getEvent({ id })
  if (!data) return null

  return (
    <EventListItem
      className={className}
      data={data}
      size={size}
      isLast={isLast}
    />
  )
}
