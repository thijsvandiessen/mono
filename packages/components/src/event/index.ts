export { Event, type Props } from './event.jsx'
