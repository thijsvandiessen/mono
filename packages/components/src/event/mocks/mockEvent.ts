import type { Event } from '@mono/graphql'

export const mockEvent: Event = {
  id: 'id',
  title: 'name',
  image: {
    id: 'image-id',
    title: 'title',
    description: 'desc',
    url: 'https://creaorkest.nl/some-image.png',
  },
  locations: [{ startTime: '2023-09-01T13:32:36+01:00', id: 'test id' }],
  url: 'some-slug',
  content: [],
}
