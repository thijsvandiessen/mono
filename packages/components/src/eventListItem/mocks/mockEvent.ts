import type { Event } from '@mono/data'

export const mockEvent: Event = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [{ startTime: '2023-09-01T13:32:36+01:00', id: 'test id' }],
  url: 'some-slug',
  content: [],
}
