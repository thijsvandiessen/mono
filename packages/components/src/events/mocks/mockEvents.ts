import type { Event } from '@mono/graphql'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  url: '/concerten/concert-1',
  image: undefined,
  locations: [],
}
