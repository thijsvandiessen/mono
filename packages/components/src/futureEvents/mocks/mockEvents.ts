import { type Event } from '@crea/graphql/src/types/event'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  image: undefined,
  locations: [],
  persons: [],
  program: [],
  url: '/some-slug',
  content: [],
}
