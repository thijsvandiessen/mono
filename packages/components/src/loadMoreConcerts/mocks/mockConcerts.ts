import { type Concert } from '@mono/data'

export const mockConcert: Concert = {
  id: 'mock-id',
  title: 'title',
  url: '/url',
  image: undefined,
  locations: [],
}
