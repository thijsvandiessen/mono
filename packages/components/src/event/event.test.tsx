import React from 'react'
import { Event, type Props } from './event'
import { getEvent } from '@mono/graphql'
import { mockEvent } from './mocks/mockEvent'
import { resolvedComponent } from '@mono/utils'
import { render, screen } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

vi.mock('../location/location', () => {
  const originalModule = vi.importActual('../location/location')
  return {
    __esModule: true,
    ...originalModule,
    Location: vi.fn(() => <p>Location Component</p>),
  }
})

vi.mock('@mono/graphql', () => {
  const originalModule = vi.importActual('@mono/graphql')
  return {
    __esModule: true,
    ...originalModule,
    getEvent: vi.fn(),
  }
})

const getEventMock = vi.mocked(getEvent)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getEventMock.mockResolvedValue({
      data: mockEvent,
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(Event, {
      id: 'some id',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()

    expect(screen.getByText('name')).toBeInTheDocument()
  })
})
