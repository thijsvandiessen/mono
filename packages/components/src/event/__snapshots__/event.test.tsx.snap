// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concert component shows all the data 1`] = `
<div>
  <div
    class="root hasImage oneLocation"
  >
    <h2
      class="title h3"
    >
      <span>
        <a
          href="some-slug"
        >
          name
        </a>
      </span>
    </h2>
    <div
      class="content"
    >
      <ul
        class="locations"
      >
        <li
          class="location"
        >
          <h3
            class="locationTitle h5"
          >
            Vrijdag 1 september â€” 14:32
          </h3>
          <p
            class="locationLinkContainer text-small"
          >
            Concert voorbij
          </p>
        </li>
      </ul>
      <div
        class="imageContainer"
      >
        <img
          alt="desc"
          class="image"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="100"
          src="/_next/image?url=https%3A%2F%2Fcreaorkest.nl%2Fsome-image.png&w=256&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fcreaorkest.nl%2Fsome-image.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fcreaorkest.nl%2Fsome-image.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
      </div>
    </div>
    <a
      class="eventLink text-small"
      href="some-slug"
    >
      Bekijk concert
      <svg
        aria-hidden="true"
        class="icon"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="currentColor"
        >
          <path
            d="M23.22 12.84H.78a.781.781 0 0 1 0-1.56h22.45c.43 0 .78.35.78.78-.01.43-.36.78-.79.78z"
          />
          <path
            d="M17.8 18.27c-.43 0-.78-.35-.78-.78 0-.21.08-.4.23-.55l4.88-4.88-4.88-4.88a.785.785 0 0 1-.03-1.1c.29-.31.79-.32 1.1-.03l.03.03 5.43 5.43c.3.3.3.79 0 1.1l-5.43 5.43a.78.78 0 0 1-.55.23z"
          />
        </g>
      </svg>
    </a>
  </div>
</div>
`;
